pytato/array.py

```diff
@@ -355,7 +355,8 @@ def normalize_shape_component(
 # {{{ array inteface
 
 SliceItem = Union[int, slice, None, EllipsisType]
-DtypeOrScalar = Union[_dtype_any, Number]
+DtypeOrScalar = Union[_dtype_any, ScalarType]
+ArrayLike = Union["Array", ScalarType]
```

Comment:
```suggestion
ArrayOrScalar = Union["Array", ScalarType]
```
?
(Numpy defines the term array-like, and I think that definition disagrees. [ref](https://numpy.org/doc/stable/reference/typing.html#module-numpy.typing))

Reply:
Indeed better not clobbering the variable name. 956b97f (Thanks!)